keras_nlp/src/models/retinanet/anchor_generator.py

```diff
@@ -0,0 +1,152 @@
+        image_shape: a list of integer numbers or Tensors representing [height,
+            width] of the input image size.
+
+    Returns:
+        multilevel_boxes: an OrderedDict from level to the generated anchor boxes of
+        shape `(H/strides[level] * W/strides[level] * len(scales) * len(aspect_ratios), 4)`.
+
+    Example:
+    ```python
+    anchor_generator = Anchor(
+        min_level=3,
+        max_level=4,
+        num_scales=2,
+        aspect_ratios=[0.5, 1., 2.],
+        anchor_size=4.,
+    )
+    anchor_generator(image_shape=(256, 256))
+    ```
+    """
+
+    def __init__(
+        self,
+        bounding_box_format,
+        min_level,
+        max_level,
+        num_scales,
+        aspect_ratios,
+        anchor_size,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.bounding_box_format = bounding_box_format
+        self.min_level = min_level
+        self.max_level = max_level
+        self.num_scales = num_scales
+        self.aspect_ratios = aspect_ratios
+        self.anchor_size = anchor_size
+        self.built = True
+
+    def call(self, images=None, image_shape=None):
+        if images is not None:
+            images_shape = images.shape
+            if len(images_shape) == 4:
+                image_shape = images_shape[1:-1]
+            else:
+                image_shape = images_shape[:-1]
+        if len(image_shape) != 2:
+            raise ValueError(
+                "Expected `image_shape` to be a Tensor of rank 2. Got "
+                f"image_shape rank={len(image_shape)}"
+            )
+        image_shape = tuple(image_shape)
+
+        multilevel_boxes = collections.OrderedDict()
+        for level in range(self.min_level, self.max_level + 1):
+            boxes_l = []
+            feat_size_y = math.ceil(image_shape[0] / 2**level)
+            feat_size_x = math.ceil(image_shape[1] / 2**level)
+
+            stride_y = ops.cast(image_shape[0] / feat_size_y, "float32")
+            stride_x = ops.cast(image_shape[1] / feat_size_x, "float32")
+
+            x = ops.arange(stride_x / 2, image_shape[1], stride_x)
+            y = ops.arange(stride_y / 2, image_shape[0], stride_y)
+
+            xv, yv = ops.meshgrid(x, y)
+
+            for scale in range(self.num_scales):
+                for aspect_ratio in self.aspect_ratios:
+                    intermidate_scale = 2 ** (scale / self.num_scales)
+                    base_anchor_size = (
+                        self.anchor_size * 2**level * intermidate_scale
+                    )
+                    aspect_x = aspect_ratio**0.5
+                    aspect_y = aspect_ratio**-0.5
+                    half_anchor_size_x = base_anchor_size * aspect_x / 2.0
+                    half_anchor_size_y = base_anchor_size * aspect_y / 2.0
+
+                    # Tensor shape Nx4
+                    boxes = ops.stack(
+                        [
+                            yv - half_anchor_size_y,
```

Comment:
what does v stand for? maybe we can change the name `yv` etc to be more readable?

Reply:
@divyashreepathihalli its basically meshgrid I will rename ito `x_grid` and `y_grid`, What do you think?

Reply:
SG!